Automattic/jetpack
# Replace deprecated e.keyCode

## Summary
This PR refactors the checks for user key press to use [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) instead of the deprecated [KeyboardEvent.keyCode](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode)

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions
The easiest way to test this is to test the Jetpack connection banner:

1. Open your jurassic tube testing site
2. Go to **/wp-admin/plugins.php**
3. Deactivate Jetpack
4. Activate Jetpack
5. You should see the following screen:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/20927667/188006311-c22cc644-7d07-49b2-bc42-03237e3087d7.png">

6. Press the **Escape** key. You should see the screen being dismissed.
